Compilation does not pass with C++23 standard

```
[28/139] Compiling C++ object src/libzim.so.9.5.0.p/fileimpl.cpp.o
FAILED: src/libzim.so.9.5.0.p/fileimpl.cpp.o 
ccache c++ -Isrc/libzim.so.9.5.0.p -Isrc -I../src -Iinclude -I../include -Istatic -fdiagnostics-color=always -Wall -Winvalid-pch -Werror -std=c++23 -O0 -g -D_LARGEFILE64_SOURCE=1 -D_FILE_OFFSET_BITS=64 -DLIBZIM_EXPORT_PRIVATE_DLL -fPIC -pthread -MD -MQ src/libzim.so.9.5.0.p/fileimpl.cpp.o -MF src/libzim.so.9.5.0.p/fileimpl.cpp.o.d -o src/libzim.so.9.5.0.p/fileimpl.cpp.o -c ../src/fileimpl.cpp
../src/fileimpl.cpp: In lambda function:
../src/fileimpl.cpp:523:52: error: implicit capture of ‘this’ via ‘[=]’ is deprecated in C++20 [-Werror=deprecated]
  523 |     auto cluster = getClusterCache().getOrPut(key, [=](){ return readCluster(idx); });
      |                                                    ^
../src/fileimpl.cpp:523:52: note: add explicit ‘this’ or ‘*this’ capture
cc1plus: all warnings being treated as errors
[37/139] Compiling C++ object src/libzim.so.9.5.0.p/writer_creator.cpp.o
ninja: build stopped: subcommand failed.
```

--- a/src/fileimpl.cpp
+++ b/src/fileimpl.cpp
@@ -520,7 +520,7 @@ private: // data
 
     auto cluster_index_type = idx.v;
     auto key = std::make_tuple(this, cluster_index_type);
-    auto cluster = getClusterCache().getOrPut(key, [=](){ return readCluster(idx); });
+    auto cluster = getClusterCache().getOrPut(key, [this, idx](){ return readCluster(idx); });
 #if ENV32BIT
     // There was a bug in the way we create the zim files using ZSTD compression.
     // We were using a too high compression level and so a window of 128Mb.

